node/chainSimulator/chainSimulator_test.go

```diff
@@ -404,3 +370,42 @@ func computeTxHash(chainSimulator ChainSimulator, tx *transaction.Transaction) (
 	txHasBytes := chainSimulator.GetNodeHandler(1).GetCoreComponents().Hasher().Compute(string(txBytes))
 	return hex.EncodeToString(txHasBytes), nil
 }
+
+func sendTxAndGenerateBlockTilTxIsExecuted(t *testing.T, chainSimulator ChainSimulator, tx *transaction.Transaction) {
+	shardID := chainSimulator.GetNodeHandler(0).GetShardCoordinator().ComputeId(tx.SndAddr)
+	err := chainSimulator.GetNodeHandler(shardID).GetFacadeHandler().ValidateTransaction(tx)
+	require.Nil(t, err)
+
+	txHash, err := computeTxHash(chainSimulator, tx)
+	require.Nil(t, err)
+	log.Warn("send transaction", "txHash", txHash)
+
+	_, err = chainSimulator.GetNodeHandler(shardID).GetFacadeHandler().SendBulkTransactions([]*transaction.Transaction{tx})
+	require.Nil(t, err)
+
+	time.Sleep(100 * time.Millisecond)
+
+	destinationShardID := chainSimulator.GetNodeHandler(0).GetShardCoordinator().ComputeId(tx.RcvAddr)
+	count := 0
+	for {
+		err = chainSimulator.GenerateBlocks(1)
+		require.Nil(t, err)
+
+		txFromMeta, errGet := chainSimulator.GetNodeHandler(destinationShardID).GetFacadeHandler().GetTransaction(txHash, true)
+		if errGet != nil && count < maxNumOfBlockToGenerateWhenExecutingTx {
+			count++
+			continue
+		}
+
+		if txFromMeta != nil && txFromMeta.Status != transaction.TxStatusPending {
+			break
+		}
+
+		count++
+		if count >= maxNumOfBlockToGenerateWhenExecutingTx {
+			t.Error("something went wrong transaction is still in pending")
+		}
+	}
+
+	log.Warn("transaction was executed", "txHash", txHash)
```

Comment:
Why warn?

Reply:
Added this Warn to check after how many blocks the transaction was executed.